### PR TITLE
Data insertion

### PR DESCRIPTION
**Pull request by dornheimer**
- map lines to conversation_id before inserting them
- clean up code for character insertion and make it more efficient (remember characters that have already been queried)
- ~~make foreign key relationship in conjunction table for characters and conversations less ambiguous~~
- ~~add alembic migration script for the new table~~

### CODE REVIEW

File: create_db.py [hunk 1]
```diff
@@ -77,15 +76,16 @@ def insert_characters(session):
     logging.info("inserted %s characters", count)
 
 
-def insert_lines(session):
+def insert_lines(session, line_to_conv):
```
Comment (fspoettel) [inline create_db.py]: `line_to_conv` could be renamed to `line_to_conv_mapping` to make it less ambiguous. `line_to_conv` could also be short for `line to convert`

**Review by fspoettel — APPROVED** [review]
Looking good and a lot more readable. Curious what the performance implications are. Did you measure?

**Comment by dornheimer** [comment]
line insertion performance increase was minor, but it also made my 'hack' unnecessary. the code is more explicit and readable now, so the change is definitely an improvement.

the real bottleneck was character insertion. last week, i timed every function in the script, but i threw away the code and didn't write down the results ;) i just timed the script as a whole and execution time on my machine was halved - from 3.07s to 1.33s

**Comment by dornheimer** [comment]
btw what is the best practice for merging? should i rebase the branch before merging to squash some commits or is that not necessary?

**Comment by fspoettel** [comment]
Squashing commits is _usually_ a good idea and can be accomplished easily via an interactive rebase. If you only want to merge one commit, github's `squash + merge` is also acceptable. You need to be careful if other people branched away from the feature branch however. Rebasing or squashing will change commit hashes and leave their branch in a conflicting state with the base branch which needs to be fixed manually via `rebase --onto` or a `cherry-pick` in some extreme cases. Can be a bit of a headache...

**Pull request merged**